docs/livepeer-in-a-box.md

```diff
@@ -0,0 +1,162 @@
+
+Thankfully, this entire process has been automated. All you need to do is have
+the project you're working on cloned in a directory adjacent to `catalyst`. For
+example, if you're hacking on `task-runner`, you might have
+
+```
+/home/user/code/catalyst
+/home/user/code/task-runner
+```
+
+The catalyst Makefile is aware of of the common paths for all of the other
+projects that go into the full stack. All that's necessary to build a new
+binary, package it in the container, and trigger a restart is a single command:
+
+```shell
+make livepeer-task-runner KILL=true
+```
+
+Note that the names of all subprojects are prefixed with `livepeer`, just like
+the resulting binaries within the Catalyst container. This yields the following
+commands:
+
+| Project                     | Command                           |
+| --------------------------- | --------------------------------- |
+| [catalyst-api]              | `make livepeer-catalyst-api`      |
+| [catalyst-uploader]         | `make livepeer-catalyst-uploader` |
+| [task-runner]               | `make livepeer-task-runner`       |
+| [analyzer]                  | `make livepeer-analyzer`          |
+| [Studio Node.js API Server] | `make livepeer-api`               |
+| [MistServer]\*              | `make mistserver`                 |
+
+\* Cross-compilation of MistServer from macOS to Linux is currently not
+supported. If you need to make changes on a Mac, you may have to use a dev
+server or a Linux VM for now.
+
+[catalyst-api]: https://github.com/livepeer/catalyst-api
+[catalyst-uploader]: https://github.com/livepeer/catalyst-uploader
+[task-runner]: https://github.com/livepeer/catalyst-uploader
+[analyzer]: https://github.com/livepeer/livepeer-data
+[Studio Node.js API Server]: https://github.com/livepeer/studio
+[MistServer]: https://github.com/livepeer/mistserver
+
+## Connecting the Frontend
+
+Livepeer in a Box comes with a [pkg](https://github.com/vercel/pkg)-bundled
+version of the Livepeer Studio API server and frontend, but does not include a
+full development environment for that frontend. If you are making changes to the
+frontend, you can boot it up as you usually would:
+
+```
+cd studio/packages/www
+yarn run dev
+```
+
+To connect it to the box; there's a hidden `localStorage` variable you can use
+to override the API server URL. Open your browser console and type in the
+following:
+
+```javascript
+localStorage.setItem("LP_API_SERVER_OVERRIDE", "http://127.0.0.1:8888");
+```
+
+Reload the page and your frontend should be connecting to the box as an API
+server.
+
+Additional note: in the interest of build speed, `make livepeer-api` does not
+package the frontend within the `livepeer-api` binary that it builds, so if you
+experience your frontend suddently 404ing after you run `make livepeer-api` you
+will have to use the above instructions to boot up the frontend on your host.
+
+## Notes
+
+-   Your CockroachDB (Postgres) database and your Minio (S3) object store will
```

Comment:
Why `CockroachDB` and not postgres `12.14` for compatibility with production?

Reply:
'cause it's in Go and I can eventually statically link it if we decide to go that route. (As opposed to linking sqlite or something like that)

Reply:
I guess that in production we'll always have a standalone database process/host. Why would be statically link it in development environment?